spec/system/admin/order_spec.rb

```diff
@@ -571,6 +571,60 @@ def new_order_with_distribution(distributor, order_cycle)
 
           expect(page).to have_content "Shipping: #{different_shipping_method_for_distributor1.name}"
         end
+
+        context "when the distributor unsupport a shipping method that's selected in an existing order " do
+          before do
+            distributor1.shipping_methods = [shipping_method_for_distributor1, different_shipping_method_for_distributor1]
+            order.shipments.each(&:refresh_rates)
+            order.shipment.adjustments.first.open
+            order.select_shipping_method(different_shipping_method_for_distributor1)
+            order.shipment.adjustments.first.close
+            distributor1.shipping_methods = [shipping_method_for_distributor1]
+          end
+          context "shipment is shipped" do
+            before do
+              order.shipments.first.update_attribute(:state, 'shipped')
+            end
+
+            it "should not change the shipping method" do
+              visit spree.edit_admin_order_path(order)
+              expect(page).to have_content "Shipping: #{different_shipping_method_for_distributor1.name} $15.00"
+            end
+
+            context "when shipping rate is updated" do
+              before do
+                different_shipping_method_for_distributor1.shipping_rates.first.update!(cost: 16)              
+              end
+
+              it "should not update the shipping cost" do
+                visit spree.edit_admin_order_path(order)
+                expect(page).to have_content "Shipping: #{different_shipping_method_for_distributor1.name} $15.00"
+              end
+            end
+          end
+          context "shipment is pending" do
+            before do
+              order.shipments.first.update_attribute(:state, 'pending')
+            end
+
+            it "should not replace the selected shipment method" do
+              visit spree.edit_admin_order_path(order)
+              expect(page).to have_content "Shipping: #{different_shipping_method_for_distributor1.name} $15.00"
```

Comment:
We're checking the UI, which is great - but maybe we could assert that the order total is not modified?

(we could do `order.reload` first and then assert on the `order.total`, but maybe there is a better way)

In fact, I think this is relevant even for each of the cases introduced in this PR.

Bonus, a tiny bit out of scope: we could add such an assertion for the case in which changing shipping methods is expected to change the order total, namely, under the block `it "can edit shipping method" do`. What do you think?

Reply:
done : )

Reply:
Thanks :-)